vault/token_store.go

```diff
@@ -917,23 +1069,187 @@ func (ts *TokenStore) destroyCubbyhole(saltedID string) error {
 
 func (ts *TokenStore) authRenew(
 	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	if req.Auth == nil {
+		return nil, fmt.Errorf("request auth is nil")
+	}
+
+	f := framework.LeaseExtend(req.Auth.Increment, 0, ts.System())
+
+	idInt, ok := req.Auth.InternalData["id"]
```

Comment:
I would just add a check to `renewAuthEntry` to pass the ClientToken through to the TokenStore. That way we can avoid tracking the extra ID.